### PR TITLE
Feature/maven central

### PR DESCRIPTION
**Pull request by marcinczeczko**
- Set up poms for release to maven central
- Set up release to maven central fat.jar, json config and logback.xml for standaolone, example-app & mocks only
- Update documentation to use resources from maven central
- Added RELEASING.md with release procedure how to guide
- Javadoc fixed to pass javadoc generation.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+$> mvn clean deploy -Prelease -Dgpg.passphrase=<your_gpg_key_passphrase>
+```
+6. On successful deploy, validate artifacts with version X.Y.Z are available on Nexus Staging:
+[https://oss.sonatype.org/content/groups/staging/io/knotx/](https://oss.sonatype.org/content/groups/staging/io/knotx/)
+7. If everything is fine, promote release to **Nexus Central Release Repositories**
+```
+$> mvn nexus-staging:release
+```
+In case you want to drop the release, you can use `nexus-staging:drop`
+9. If everything is fine, promote release to **Nexus Central Release Repositories**
```
Comment (Skejven) [inline RELEASING.md]: Missing point (8). And it looks that something here is wrong in point 9.

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+##Prerequisites
+1. Sonatype.org [JIRA](https://issues.sonatype.org/secure/Signup!default.jspa) account
+
+2. Your Sonatype.org account added to Knot.x project (if not, please contact Knot.x team: 
+[knotx.team@gmail.com](email:knotx.team@gmail.com))
+
+3. You have create GPG key for the email you have registered on Sonatype.org JIRA 
+(Follow the [Working with PGP Signatures](http://central.sonatype.org/pages/working-with-pgp-signatures.html) 
+guide how to do it). 
+**Don't forgot to deploy your public key to keyserver**
```
Comment (marcinczeczko) [inline RELEASING.md]: fixed

File: RELEASING.md (hunk 1 above)
Comment (marcinczeczko) [inline RELEASING.md]: fixed

**Comment by marcinczeczko** [comment]
@toniedzwiedz - today evening is fine.

**Pull request merged**